app/api/endpoint.py

```
@@ -0,0 +1,63 @@
+router = APIRouter()
+
+@router.get("/recommend-user")
+def recommend_on_demand(request: Request):
+    try:
+        # 0. 쿠키에서 user_id 추출
+        try:
+            user_id = get_current_user_id_from_token(request)
+            print(f"[추천 API] AccessToken에서 추출된 user_id: {user_id}")
+        except Exception as e:
+            print(f"[ERROR] user_id 추출 실패: {e}")
+            raise HTTPException(status_code=401, detail="Access token이 유효하지 않습니다.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)